dask/array/core.py

```diff
@@ -2009,6 +2019,83 @@ def from_array(x, chunks, name=None, lock=False, asarray=True, fancy=True,
     return Array(dsk, name, chunks, dtype=x.dtype)
 
 
+def from_zarr(url, component=None, storage_options=None, **kwargs):
+    """Load array from the zarr storage format
+
+    See https://zarr.readthedocs.io for details about the format.
+
+    Parameters
+    ----------
+    url: str
+        Location of the data. This can include a protocol specifier like s3://
+        for remote data.
```

Comment:
I wonder if it's worth allowing this first argument to either be a string (in which case interpreted as url) or a mapping. Then you'd get full generality to use any type of store, including zip files, lmdb, etc.

Reply:
A suitable mapping is any instance of `collections.MutableMapping`?

Reply:
That sounds correct (relevant doc snippet below). There are some operations that can be optimized by defining some special internal methods, but they are not strictly required.

> Note that any object implementing the `MutableMapping` interface from the `collections` module in the Python standard library can be used as a Zarr array store, as long as it accepts string (str) keys and bytes values.

ref: http://zarr.readthedocs.io/en/latest/api/storage.html#module-zarr.storage

Reply:
That said, have seen a lot of work that you have been doing involving URLs and paths in the Dataframe's side of the code base (though am not to familiar with what is going on there). Maybe it's worth mentioning what you had in mind and how that compares to what has been done with Dataframe's already.

Reply:
FWIW I think you could check if the argument is a string, if so interpret as URL, otherwise assume it's a mapping (let duck typing occur naturally by passing through to zarr). But if you want an explicit check, all store classes in zarr do sub-class from MutableMapping, so should be fine too.

Reply:
I think the first of those two suggestions makes the most sense to me.